easyfsl/samplers/task_sampler.py

```diff
@@ -92,18 +96,54 @@ def episodic_collate_fn(
             [true_class_ids.index(x[1]) for x in input_data]
         ).reshape((self.n_way, self.n_shot + self.n_query))
         # pylint: enable=not-callable
-
         support_images = all_images[:, : self.n_shot].reshape(
             (-1, *all_images.shape[2:])
         )
         query_images = all_images[:, self.n_shot :].reshape((-1, *all_images.shape[2:]))
         support_labels = all_labels[:, : self.n_shot].flatten()
         query_labels = all_labels[:, self.n_shot :].flatten()
-
         return (
             support_images,
             support_labels,
             query_images,
             query_labels,
             true_class_ids,
         )
+    def check_episodic_collate_fn_input(
+        self, input_data: List[Tuple[Tensor, Union[Tensor, int]]]
+    ) -> bool:
+        """
+        Check the type of the input for the episodic_collate_fn method.
+        Args:
+            input_data: each element is a tuple containing:
+                - an image as a torch Tensor
+                - the label of this image as an int or a 0-dim tensor
+        Returns:
+            bool,bool: respectively::
+                - True if the input is of correct type, False if input is of wrong type.
+                - True if the input labels needs to be transformed from tensor to int and False otherwise.
+        """
+        #that flag is true when the input labels are  0-dim tensors
+        o_dim_tesors_labesl=False
+        for img, label in input_data:
+            #if that label isn't int then check if it's a 0-tensor
+            if not  (isinstance(img,torch.Tensor) and isinstance(label,int)):
+                if not (isinstance(img,torch.Tensor) and isinstance(label,torch.Tensor) and label.ndim==0):
```

Comment:
(!) You're checking `img`'s type twice. I suggest a cleaner case switch, something like this:

```python
for item_index, (image, label) in enumerate(input_data):
	if not isinstance(image, Tensor): 
		raise TypeError("...")
	if not isinstance(label, int):
		if not isinstance(label, Tensor):
			raise TypeError("...")
		if label.ndim !=0:
			raise ValueError("...")
		input_data[item_index] = (image, int(label))
```